Inspector: Implemented a collapse/expand all

Krakean opened the pull request:
Implemented a collapse all / expand all feature request (fixes #9427)
This is how it works in action: https://i.imgur.com/RTmLaPh.gifv

Pretty small addition, hopefully will be added to 3.0 :)

P.S. Thought about to implement it as one-switch button, but it would require a much of unnecessary changes for 3.0-beta considering that we are in feature freeze state. Current solution is pretty simple and neat, in my opinion. And do not introduce any issues :)

Krakean committed:
Implemented a collapse/expand all feature request for Inspector (issue #9427).

add dot to end of hints as suggested by Calinou

akien-mga commented on editor/property_editor.h:
```diff
@@ -1,4 +1,4 @@
-﻿/*************************************************************************/
+/*************************************************************************/
```
Your IDE added a BOM here, could you amend the commit to remove it?

Krakean replied:
Can you please manually fix it? Can't see a BOM, nor in Visual Studio nor in Notepad++ nor in AkelPad. Shows me typical UTF-8 without any bom.

Krakean replied:
And clang-format also doesn't fix anything.

akien-mga replied:
Nevermind, it looks like you *removed* an existing BOM, so that's good.

Krakean committed:
Merge branch 'master' of https://github.com/godotengine/godot into add_editorinspector_collapseexpand

djrm commented:
im not sure about this one, two buttons dont seem right, i think it would be better to add a context menu with `expand all` and `collapse all` to tree, that way every tree will benefit from this. (the context menu could be activated by right clicking on section titles or something like that)

Krakean commented:
@djrm 
Context menu doesn't seems right to me, it adds one more mouse click which is unneeded in this situation in my opinion. 
If you think that two buttons doesn't seems right anyway, well, one button is not solution too. 
Simple case: https://i.imgur.com/7gCjCfk.png -- what icon I should display in this case, when just one or two properties are expanded -- expand all or collapse all?

djrm commented:
i never said one button that doesnt makes sense, what i was suggesting was a less intrusive more standarized way to access such functionality, it would be useful in other trees, like input map, also one more press is not bad for a secondary action like this (and its already saving you a ton of presses).

Krakean commented:
@djrm 
> standarized way to access such functionality, it would be useful in other trees, like input map

Will think about adding such funcitonality to other places too :)
But first of all would be nice if this PR will be merged, as for start.

Dillybob92 commented:
I was thinking of global setting that we change in Editor Settings. Either collapse inspectors or not (Boolean), but I mean to get this before feature freeze would be ideal too :D Awesomeeee!!

Krakean commented:
@Dillybob92 current my implementation of this feature allows easily to add an Editor Settings-level option, so this is something I keep in mind if  "collapse/expand all" feature will be actually used & welcomed by users in general.

reduz commented:
I think this should be implemented as follows:

1. Move the buttons to the menu:

![image](https://user-images.githubusercontent.com/6265307/33135929-a18cd818-cf82-11e7-9c88-653c6098745b.png)

2. Add a project setting to make property editor open unfolded on default (if this does not exist already)

The rationale is that, if you will be opening this all the time, you may as well make it the default. If you will do it ocassionally, then you don't need a button visible all the time.

Krakean commented:
@reduz
1. Thought about that, but decided to position them in order of importance from low to highest, as I understand importance of "show search" and "object properties" buttons. But anyway, will do as you say, not a problem.
2. Okay, will add an editor setting for it. Didn't made it initially due to feature freeze state, so thought PR could be not accepted if I add more than necessary. 

PR will be updated in two days, at the moment my PC in service center.

groud commented:
From what I understand, feature freeze does not include usability improvements. So it's ok merging this :)

reduz commented:
@Krakean you have until the weekend for beta, afterwards we won't merge any more PRs that are not fixes

Krakean commented:
@reduz
Got it.

Krakean committed:
Merge branch 'master' of https://github.com/godotengine/godot into add_editorinspector_collapseexpand

Krakean committed:
Implemented a collapse/expand all feature request for Inspector (issue #9427) via popup of "Object properties" button.
Editor Settings->Interface->Editor: added "Expand All Properties" option. Off by default.

Krakean commented:
Reimplemented PR as #13233

akien-mga commented:
Superseded by #13233.
